test/unit/data/test_galaxy_mapping.py

```diff
@@ -472,6 +472,25 @@ def contents_iter_names(**kwds):
 
         assert contents_iter_names(ids=[d1.id, d3.id]) == ["1", "3"]
 
+    def test_history_audit(self):
```

Comment:
Thanks for adding this! Last suggestion: we're testing that we prune records, but not that we prune the correct ones (so the query logic is only partially covered). If you think this is worth addressing, I think a simple modification to the test can do it, for example: we can have h1 and h2; change `count_audit_table_entries()` to `get_audit_table_entries(h)`, call it for both h1 and h2, then extract `max` update_time for both after the call after flush. Then we can verify that the correct records are pruned for number-of-histories > 1.

Reply:
I'm almost sure I gave you access to the branch if you want to just mod it yourself. Thanks to both of you guys for doing the stuff i had no idea how to approach.

Reply:
done.

Reply:
Awesome